gprofiler/application_separators.py

```diff
@@ -0,0 +1,193 @@
+import configparser
+import os.path
+import re
+from abc import ABCMeta, abstractmethod
+from typing import List, Optional
+
+from granulate_utils.linux.ns import resolve_host_path
+from psutil import NoSuchProcess, Process
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+class ApplicationSeparator(metaclass=ABCMeta):
```

Comment:
Are theses classes part of the API of the module? If not they should be prefixed with `_`. It'll make it much easier to understand how this module is supposed to be used.
Also, it may fit better under `gprofiler/metadata/`.

Reply:
They are not a part of the API of the module, but It's really weird to prefix classes with `_` it's common to have classes that are not API of the module without a prefix.

Reply:
I don't think it's weird. Actually, I believe it's the standard (also according to PEP8).
You can search for `class _` in cpython.

Reply:
CPython is not fully following PEP8, and PEP8 only talks about **private attributes** the convention of function names starting with an underscore is not a part of the PEP8 convention.

Reply:
From [PEP 8](https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces):
> Even with \_\_all\_\_ set appropriately, internal interfaces (packages, modules, classes, functions, attributes or other names) should still be prefixed with a single leading underscore.

Reply:
I don't see a reason not to stick to PEP 8. Conventions are important for readability, especially when they're so widely excepted like PEP 8 and in an open source project

Reply:
kk, agreed with @YishaiZinkin fixed it and moved to `metadata` dir